platform/osx/export/export_plugin.cpp

```diff
@@ -435,6 +435,52 @@ Error EditorExportPlatformOSX::_code_sign(const Ref<EditorExportPreset> &p_prese
 	return OK;
 }
 
+Error EditorExportPlatformOSX::_code_sign_directory(const Ref<EditorExportPreset>& p_preset, const String& p_path,
+                                                    const String& p_ent_path) {
+#ifdef OSX_ENABLED
+    static Vector<String> extensions_to_sign;
+
+    if (extensions_to_sign.is_empty()) {
+        extensions_to_sign.push_back("dylib");
+        extensions_to_sign.push_back("framework");
+    }
+
+    Error dir_access_error;
+    DirAccessRef dir_access{DirAccess::open(p_path, &dir_access_error)};
+
+    if (dir_access_error != OK) {
+        return dir_access_error;
+    }
+
+    dir_access->list_dir_begin();
+    String current_file{dir_access->get_next()};
+    while (!current_file.is_empty()) {
+        String current_file_path{p_path.plus_file(current_file)};
+
+        if (current_file == ".." || current_file == ".") {
+            current_file = dir_access->get_next();
+            continue;
+        }
+
+        if (extensions_to_sign.find(current_file.get_extension()) > -1) {
```

Comment:
Just checking for `!current_file..get_extension().is_empty()` should be enough, nothing except dylibs and frameworks is allowed to be in the `Frameworks` anyway.